lib/FFI/Platypus/Type/Win32/LPCWSTR.pm

```diff
@@ -0,0 +1,132 @@
+package FFI::Platypus::Type::Win32::LPCWSTR;
+
+use strict;
+use warnings;
+use 5.008004;
+use FFI::Platypus;
+use FFI::Platypus::Memory qw( strlenW );
+use Encode qw( decode encode );
+
+# ABSTRACT: Platypus custom type for Windows Unicode strings
+# VERSION
+
+=head1 SYNOPSIS
+
+In your C code:
+
+ void
+ takes_unicode_string(LPCWSTR s)
+ {
+   ...
+ }
+
+In your L<Platypus::FFI> code:
+
+ use FFI::Platypus;
+
+ my $ffi = FFI::Platypus->new( api => 1 );
+ $ffi->load_custom_type('::Win32::LPCWSTR' => 'LPCWSTR');
+
+ $ffi->attach(takes_unicode_string => ['LPCWSTR'] => 'void');
+
+ my $str = "I ❤️ Platypus";
+ takes_unicode_string($str);
```

Comment:
Maybe we can use `MessageBoxExW` for the example here instead of including our own C code.

https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-messageboxexw

Reply:
or actually `MessageBoxW` is less complicated

https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-messageboxw

Reply:
You want the test suite to popup a dialog and wait for the user to close it??? Smokers won't be able to run this, for one.

Reply:
argh, this is just documentation. Sorry, just rushing a reply.